packages/superset-ui-dimension/src/getTextDimension.ts

```diff
@@ -17,39 +10,27 @@ export interface GetTextDimensionInput {
   text: string;
 }
 
-const DEFAULT_DIMENSION = { height: 20, width: 100 };
-
 export default function getTextDimension(
   input: GetTextDimensionInput,
-  defaultDimension: Dimension = DEFAULT_DIMENSION,
+  defaultDimension?: Dimension,
 ): Dimension {
-  const { text, className, style = {}, container = document.body } = input;
+  const { text, className, style, container } = input;
 
-  const textNode = document.createElementNS(SVG_NS, 'text');
-  textNode.textContent = text;
-
-  if (className !== undefined && className !== null) {
-    textNode.setAttribute('class', className);
+  // Empty string
+  if (text.length === 0) {
+    return { height: 0, width: 0 };
   }
 
-  STYLE_FIELDS.filter(
-    (field: keyof TextStyle) => style[field] !== undefined && style[field] !== null,
-  ).forEach((field: keyof TextStyle) => {
-    textNode.style[field] = `${style[field]}`;
-  });
-
-  const svg = document.createElementNS(SVG_NS, 'svg');
-  svg.style.position = 'absolute'; // so it won't disrupt page layout
-  svg.style.opacity = '0'; // and not visible
-  svg.style.pointerEvents = 'none'; // and not capturing mouse events
-  svg.appendChild(textNode);
-  container.appendChild(svg);
+  const svgNode = hiddenSvgFactory.createInContainer(container);
+  const textNode = textFactory.createInContainer(svgNode);
+  updateTextNode(textNode, { className, style, text });
+  const dimension = getBBoxCeil(textNode, defaultDimension);
 
-  const bbox = textNode.getBBox ? textNode.getBBox() : defaultDimension;
-  container.removeChild(svg);
+  setTimeout(() => {
```

Comment:
The nodes are added to the DOM briefly only to make `getBBox` works. (If not added to DOM `getBBox` will always return `0,0`) After that the svg nodes are not needed. We delay its removal in case there are subsequent `getTextDimension` calls that can reuse the svg nodes. Experiments by @betodealmeida shown that reusing existing nodes instead of deleting and adding new ones can save lot of time.